src/user/user.service.ts

```diff
@@ -20,13 +20,13 @@ export class UserService {
     return this.userRepository.findAll();
   }
 
-  async findOne(loginUserDto: LoginUserDto): Promise<User> {
+  async findOne(loginUserDto: LoginUserDto): Promise<User|null> {
```

Comment:
```suggestion
  async findOne(loginUserDto: LoginUserDto): Promise<User | null> {
```

Reply:
OK. Why are the tests failing by the way? And what other refactoring can potentially be done to the project?

Reply:
because the return values are mocked 

https://github.com/mikro-orm/nestjs-realworld-example-app/blob/master/src/user/test/user.service.spec.ts#L24

Reply:
I don't see the problem here, since I did not touch the tests code, and the mocked returned object is the same for `findOne()` and `findOneOrFail()`

Reply:
You are right. But that's the problem of test mocks implementation. In fact all of that objects should contain a password field; there is a custom method `createUserRO()` that strips out some fields including the password one, but the real `findOneOrFail()` EntityManager method surely does not strip anything.

Reply:
yes, as i said earlier, the implementation could deserve a bigger refactor, this doesn't need custom handling, the password should be rather marked as `lazy` and optionally fetched only when needed